rskj-core/src/main/java/co/rsk/cli/tools/ExecuteBlocks.java

```diff
@@ -46,7 +49,12 @@ public static void execute(String[] args, BlockExecutor blockExecutor, BlockStor
             Block block = blockStore.getChainBlockByNumber(n);
             Block parent = blockStore.getBlockByHash(block.getParentHash().getBytes());
 
-            blockExecutor.execute(block, parent.getHeader(), false, false);
+            BlockResult blockResult = blockExecutor.execute(block, parent.getHeader(), false, false);
+
+            if (!Arrays.equals(blockResult.getFinalState().getHash().getBytes(), block.getStateRoot())) {
```

Comment:
Important warning about specifically this change on #1526 so we don't miss it. This will fail on old blocks (number < 1591000)